src/main/java/com/altenheim/kalender/controller/logicController/ImportExportTest.java

```diff
@@ -0,0 +1,164 @@
+            var summary = ((Property) components.get(i).getProperties().getProperty("SUMMARY")).getValue();
+            entry.setTitle(summary);
+            var locationProp = (Property) components.get(i).getProperties().getProperty("LOCATION");
+            if(locationProp != null)
+            {
+                entry.setLocation(locationProp.getValue());
+            }
+            cal.addEntry(entry);
+        }
+        return cal;
+    } 
+
+    public void exportFile(com.calendarfx.model.Calendar cal) throws ValidationException, IOException
+    {
+        var entries = cal.findEntries("");
+        var calName = cal.getName();
+        var icsCalendar = new Calendar();
+        icsCalendar.getProperties().add(new ProdId("-//Smart Planner//iCal4j 1.0//DE"));
+        icsCalendar.getProperties().add(Version.VERSION_2_0);
+        icsCalendar.getProperties().add(CalScale.GREGORIAN);
+        icsCalendar.getProperties().add(new XProperty("X-WR-CALNAME", calName));
+        for(int i = 0; i < entries.size(); i++)
+        {
+            var entry = entries.get(i);
+            icsCalendar.getComponents().add(createIcalEntryFromCalFXEntry(entry));
+        }
+        var fout = new FileOutputStream(calName + ".ics");
+        var outputter = new CalendarOutputter();
+        outputter.output(icsCalendar, fout);
+        System.out.println("File created in root directory.");
+    }
+
+    public void exportFile(com.calendarfx.model.Calendar[] cal) throws ValidationException, IOException
+    {
+        var test = true;
+        for(int i = 0; i < cal.length; i++)
+        {
+            var entries = cal[i].findEntries("");
+            var calName = cal[i].getName();
+            var icsCalendar = new Calendar();
+            icsCalendar.getProperties().add(new ProdId("-//Smart Planner//iCal4j 1.0//DE"));
+            icsCalendar.getProperties().add(Version.VERSION_2_0);
+            icsCalendar.getProperties().add(CalScale.GREGORIAN);
+            icsCalendar.getProperties().add(new XProperty("X-WR-CALNAME", calName));
+            for(int j = 0; j < entries.size(); j++)
+            {
+                var entry = entries.get(j);
+                icsCalendar.getComponents().add(createIcalEntryFromCalFXEntry(entry));
+            }
+            var path = calName + ".ics";
+            var fout = new FileOutputStream(path);
+            var outputter = new CalendarOutputter();
+            outputter.output(icsCalendar, fout);
+            var file = new File(path);
+            if(!file.exists())
+                test = false;
+        }
+        if(test)
+            System.out.println("Files created in root directory.");
+        else
+            System.out.println("An error has occurred.");
+    }
+
+    public void exportFile(Entry<?> ent) throws ValidationException, IOException
+    {
+        var icsCalendar = new Calendar();
+        icsCalendar.getProperties().add(new ProdId("-//Smart Planner//iCal4j 1.0//DE"));
```

Comment:
Fällt was auf? Diesen Block verwendet ihr jetzt zum 3. Mal. Ruft doch diese Methode in den anderen lieber auf statt dasselbe innerhalb der Methoden zu wiederholen zu wiederholen. DRY Prinzip: Dont repeat yourself. Wenn (programmweit!) Code doppelt vorkommt stimmt was nicht.  Input bleibt ja sogar entry, Rückgabe könnte der Outputter sein.